File: GitNexus/gitnexus/test/fixtures/lang-resolution/cpp-deep-field-chain/service.cpp
#include "models.h"

void processUser(User user) {
    // 2-level chain: user.address → Address, then .save() → Address#save
    user.address.save();

    // 3-level chain: user.address → Address, .city → City, .getName() → City#getName
    user.address.city.getName();
}

File: GitNexus/gitnexus/test/fixtures/lang-resolution/field-types/service.ts
import { User, Config } from './models';

function processUser(user: User) {
  // Field-access chain: user.address resolves to Address, then .save() resolves to Address#save
  user.address.save();
}

function validateConfig() {
  // Static field access: Config.DEFAULT resolves to Config, then .validate() resolves to Config#validate
  Config.DEFAULT.validate();
}
